src/states/Game.gd

```diff
@@ -56,6 +62,26 @@ func reset_move_counter() -> void:
 	update_move_counter()
 
 
+func update_timer(final_time := false) -> void:
+	var should_show_timer := Autosplitter.speedrun_is_running and Global.speedrun_timer_enabled
+	if not should_show_timer == speedrun_timer.visible:
+		speedrun_timer.visible = should_show_timer
+
+	var minutes := floor(Autosplitter.speedrun_time / 60.0) as int
+	var non_minutes := fmod(Autosplitter.speedrun_time, 60.0)
+	var seconds := floor(non_minutes) as int
+	var non_seconds := fmod(non_minutes, 1.0)
+	var milliseconds := floor(non_seconds * 1000) as int
+
+	var time_str = "%02d.%03d" % [seconds, milliseconds]
+	if minutes > 0:
+		time_str = "%d:%s" % [minutes, time_str]
+	if final_time:
+		time_str = "%s Final Time" % time_str
```

Comment:
I wonder if we want to make the text change color (green while running, while when complete? or smth)

Reply:
I thought about it, but I don't think it's a high priority at least for this PR